contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c

```diff
@@ -564,18 +564,31 @@ resolve_numeric_typmod_from_exp(Node *expr)
 			 * Otherwise it simply won't fit in 38 precision and let an
 			 * overflow error be thrown in PrepareRowDescription.
 			 */
-			if (precision > TDS_MAX_NUM_PRECISION &&
-			    precision - scale <= TDS_MAX_NUM_PRECISION)
+			if (precision > TDS_MAX_NUM_PRECISION)
 			{
-			    /*
-			     * scale adjustment by delta is only applicable for division
-			     * and (multiplcation having no aggregate operand)
-			     */
-			    int delta = precision - TDS_MAX_NUM_PRECISION;
-			    precision = TDS_MAX_NUM_PRECISION;
-			    scale = Max(scale - delta, 0);
+				if (precision - scale > 32 && scale > 6)
+				{
+					/*
+					 * Result might be rounded to 6 decimal places or the overflow error
+					 * will be thrown if the integral part can't fit into 32 digits.
```

Comment:
where will we detect this overflow if intergral part can't fit in 38 places? Also, why we are only limiting scale to 6? what if intergral part required more than 32 palces

Reply:
Q: where will we detect this overflow if intergral part can't fit in 38 places?
A: When integral part can't be handled in 38 places, we throw overflow error based on actual numeric data length. [This part was added in BABEL-3179[https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/pull/40 and https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/495]

Q: Also, why we are only limiting scale to 6? what if intergral part required more than 32 places?
A. We are following sql server behaviour, where if (precision - scale > 32 && scale > 6), we set scale to 6 and if integral part can't be accommodated in rest of 32 digits, overflow error is triggered from TDS side.
Followed behaviour: https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/t-sql/data-types/precision-scale-and-length-transact-sql.md